docs/weave-tutorial/07-handlers.md

```diff
@@ -0,0 +1,675 @@
+
+First of all, we implement the `validate` function that validates the message with static checks:
+
+```go
+// validate does all common pre-processing between Check and Deliver
+func (h CreateBlogHandler) validate(ctx weave.Context, store weave.KVStore, tx weave.Tx) (*CreateBlogMsg, *Blog, error) {
+    var msg CreateBlogMsg
+
+    if err := weave.LoadMsg(tx, &msg); err != nil {
+        return nil, nil, errors.Wrap(err, "load msg")
+    }
+
+    blockTime, err := weave.BlockTime(ctx)
+    if err != nil {
+        return nil, nil, errors.Wrap(err, "no block time in header")
+    }
+    now := weave.AsUnixTime(blockTime)
+
+    blog := &Blog{
+        Metadata:    msg.Metadata,
+        Owner:       x.MainSigner(ctx, h.auth).Address(),
+        Title:       msg.Title,
+        Description: msg.Description,
+        CreatedAt:   now,
+    }
+
+    return &msg, blog, nil
+}
+```
+
+Let's go over the `validate` method piece by piece:
+
+1. Message is validated with static checks such as missing title check
+2. Execution time of the transaction which will be used as creation time of the blog, is extracted from `BlockTime` info that lives in the `Context` of the handler.
+3. `Blog` instance is created with the values taken from `msg`
+
+I did not mention the error checks between these steps but they are obvious and intuitive yet very important. Go does good job with forcing developer to pay attention to errors.
+
+Then implement `Check` method that checks constraints like if **fee** is sufficent to execute the transaction and transactions is eligible to enter the mempool using validation:
+
+```go
+// Check verifies it is properly formed and returns
+// the cost of executing it.
+func (h CreateBlogHandler) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx) (*weave.CheckResult, error) {
+    _, err := h.validate(ctx, store, tx)
+    if err != nil {
+        return nil, err
+    }
+
+    return &weave.CheckResult{GasAllocated: newBlogCost}, nil
+}
+```
+
+`newBlogCost` is defined as 10 tokens.
+
+```go
+// Deliver creates an custom state and saves if all preconditions are met
+func (h CreateBlogHandler) Deliver(ctx weave.Context, store weave.KVStore, tx weave.Tx) (*weave.DeliverResult, error) {
+    _, blog, err := h.validate(ctx, store, tx)
+    if err != nil {
+        return nil, err
+    }
+
+    err = h.b.Put(store, blog)
+    if err != nil {
+        return nil, errors.Wrap(err, "cannot store blog")
+    }
+
+    // Returns generated blog ID as response
+    return &weave.DeliverResult{Data: blog.ID}, nil
+}
+```
+
+Noticed `validate` method is also used in the beginning of `Deliver` execution? Purpose of this is to prevent any message manipulation between `Check` and `Deliver` methods. We would not want to any malformed state changes to be applied in the blockchain.
```

Comment:
```suggestion
Notice that `validate` method is also used in the beginning of `Deliver` execution? The purpose of this is to prevent any message manipulation between `Check` and `Deliver` methods. We would not want to any malformed state changes to be applied in the blockchain.
```